plugins/kotlin/jvm-debugger/util/src/org/jetbrains/kotlin/idea/debugger/InlineUtils.kt

```diff
@@ -147,10 +153,11 @@ class InlineStackFrameVariableHolder private constructor(
                     // point, since arguments to an inline function argument would be
                     // associated with a previous active frame.
                     //
-                    // If we do encounter pending variables, then they belong to an
-                    // inline call whose scope introduction variable was shadowed by
-                    // a later call to the same function and can be hidden.
+                    // If we do encounter pending variables, then this indicates that the
+                    // debug information is inconsistent and we mark all pending variables
+                    // as invalid.
                     name.startsWith(LOCAL_VARIABLE_NAME_PREFIX_INLINE_ARGUMENT) -> {
+                        require(pendingVariables.isEmpty())
```

Comment:
What is this line needed for? Because of it, the code below doesn't make sense. Previously there was an assert here, but we decided to invalidate pending variables instead of throwing an exception.

Reply:
Good catch, I added some requires for testing that these workarounds are no longer needed and forgot to take this one out. Fixed!